### ISSUE TITLE
Mocked S3 client does not support more than 1000 objects

### ISSUE THREAD

**Issue by davidjmemmett** [getmoto/moto#7286]
I've recently upgraded a project from Python 3.8.2 to 3.12.1, which then pulled in a later version of moto (v5.0.0) (noticeable change was `mock_s3` is now just `mock_aws`).

After running my tests, one failed which attempted to list over 1000 objects. Please find below a snippet which reproduces the error:

```python
import boto3
import moto
from io import BytesIO

mock_s3 = moto.mock_aws()
mock_s3.start()
s3 = boto3.client('s3')
s3.create_bucket(Bucket='test')


for i in range(1050):
    s3.upload_fileobj(Fileobj=BytesIO(), Bucket='test', Key=f'key{i}')

response = s3.list_objects_v2(Bucket='test')
print(response.keys())

print(response['IsTruncated'])
```

Note that 1050 files are uploaded, yet the response contains 1000 items, and `IsTruncated` is `False`.

The same tests passed using the previous version of moto that was installed in Python 3.8.2 (apologies, I don't know what version of moto this was).

I have repeated the same test (changing `mock_aws` back to `mock_s3` using Python 3.12.1 and moto~=4.2 (which installed 4.2.14) and it now behaves as expected (`IsTruncated` returns `True`).

**Comment by bblommers** [comment]
Hi @davidjmemmett, this should be fixed as of moto >= 5.0.1.dev19. Are you able to upgrade and verify that works for you?

**Comment by davidjmemmett** [comment]
Thank you for the prompt response @bblommers - I can confirm that this is fixed as of `moto==5.0.1.dev19`
